### PR TITLE
chore: introduce vitests for tests

### PR DESCRIPTION
**Pull request by danez**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

This introduces vitests for test files with mjs extension.

For integration tests the CI uses sharding again to split up tests into multiple workers, right now with one test this is kind of useless, but will be important later. In PRs the flag `--changed` is used to only run tests that have changes. 

vitest is clever enough to rerun all tests if package.json and shrinkwrap change. :)

The Jobs have also been renamed and are now called

- `Tests / E2E`
- `Tests / Unit`
- `Tests / Integration`
- `Tests / Legacy Integration` (which will be going away)

I was wondering what the `all` [job](https://github.com/netlify/cli/blob/main/.github/workflows/main.yml#L150-L155) is doing and found this: https://github.com/netlify/cli/pull/4164#discussion_r795925906
So it was meant to be the only job required from this workflow, which isn't the case right now as all test jobs are required. I opted to remove the all job for that reason, but happy to restore.

Ref #5265 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/lib/edge-functions/consts.cjs [hunk 1]
```diff
@@ -1,21 +1,11 @@
...
-const SERVER_POLL_TIMEOUT = 1e4
-
 module.exports = {
-  DEFAULT_SRC_DIR,
   DIST_IMPORT_MAP_PATH,
   INTERNAL_EDGE_FUNCTIONS_FOLDER,
   EDGE_FUNCTIONS_FOLDER,
   PUBLIC_URL_PATH,
-  SERVER_POLL_INTERNAL,
-  SERVER_POLL_TIMEOUT,
```
Comment (danez) [inline src/lib/edge-functions/consts.cjs:20]: All unused

File: e2e.config.mjs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const config = {
   files: ['e2e/**/*.e2e.mjs'],
   cache: true,
-  // eslint-disable-next-line no-magic-numbers
+
```
Comment (jackiewmacharia) [inline e2e.config.mjs:4]:
```suggestion
```
(nitpick/dust) suggestion removes extra empty line

File: tests/integration/utils/got.cjs [hunk 3]
```diff
@@ -13,7 +13,7 @@ const isResponseOk = (response) => {
 const TIMEOUT = 3e5
 
 // Default got retry status code with the addition of 403
-// eslint-disable-next-line no-magic-numbers
+
```
Comment (jackiewmacharia) [inline tests/integration/utils/got.cjs:16]:
```suggestion
```
(nitpick/dust) suggestion removes extra empty line

File: tests/unit/lib/functions/server.test.cjs [hunk 4]
```diff
@@ -23,7 +23,7 @@ test.before(async () => {
     projectRoot,
     config: {},
     timeouts: { syncFunctions: 1, backgroundFunctions: 1 },
-    // eslint-disable-next-line no-magic-numbers
+
```
Comment (jackiewmacharia) [inline tests/unit/lib/functions/server.test.cjs:26]:
```suggestion
```
(nitpick/dust) suggestion removes extra empty line

**Review by jackiewmacharia — COMMENTED** [review]
I've added a few nitpick suggestions, all in all, this looks good.

**Review by jackiewmacharia — APPROVED** [review]

**Review by jackiewmacharia — DISMISSED** [review]

**Review by khendrikse — APPROVED** [review]

**Pull request merged**